addons/purchase/models/purchase.py

```diff
@@ -469,6 +469,13 @@ def button_draft(self):
 
     def button_confirm(self):
         for order in self:
+            if self.env.context.get('validate_analytic', False):
```

Comment:
Should this not go after the state check below (and in a method as mentioned above)